<a href="https://colab.research.google.com/github/arsitapraktik-ctrl/data-science-2026/blob/main/Pertemuan2%5BArsita%5D%5B240401010318%5D_ipynb.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [2]:
import pandas as pd
import numpy as np

In [3]:
url = "https://raw.githubusercontent.com/datasciencedojo/datasets/master/titanic.csv"
df = pd.read_csv(url)

df.head()

,PassengerId,Survived,Pclass,Name,Sex,Age,SibSp,Parch,Ticket,Fare,Cabin,Embarked
0,1,0,3,"Braund, Mr. Owen Harris",male,22.0,1,0,A/5 21171,7.2500,NaN,S
1,2,1,1,"Cumings, Mrs. John Bradley (Florence Briggs Th...",female,38.0,1,0,PC 17599,71.2833,C85,C
2,3,1,3,"Heikkinen, Miss. Laina",female,26.0,0,0,STON/O2. 3101282,7.9250,NaN,S
3,4,1,1,"Futrelle, Mrs. Jacques Heath (Lily May Peel)",female,35.0,1,0,113803,53.1000,C123,S
4,5,0,3,"Allen, Mr. William Henry",male,35.0,0,0,373450,8.0500,NaN,S


Q1: Berapa jumlah baris dan kolom dataset titanic? kolom apa saja yang tersedia?

In [4]:
print(df.shape)
print(df.columns)

(891, 12)
Index(['PassengerId', 'Survived', 'Pclass', 'Name', 'Sex', 'Age', 'SibSp',
       'Parch', 'Ticket', 'Fare', 'Cabin', 'Embarked'],
      dtype='object')


Data set titanic memiliki 891 baris dan 12 kolom yang merepresentasikan informasi penumpang seperti kelas, jenis kelamin, usia, dan status keselamatan.

Q2: kolom mana saja yang memiliki missing values? Berapa jumlah dan proporsinya (%)

In [5]:
(df.isnull().sum() / len(df)) * 100

,0
PassengerId,0.000000
Survived,0.000000
Pclass,0.000000
Name,0.000000
Sex,0.000000
Age,19.865320
SibSp,0.000000
Parch,0.000000
Ticket,0.000000
Fare,0.000000


Terdapat missing value pada beberap kolom, terutama Cabin (687 data kosong), Age 177, dan Embarked. Kolom Cabin memiliki jumlah data kosong peling bnyak sehingga kemungkinan perlu di hapus atau tidak di gunakan dalam analisis

Q3: Berapa persen penumpang yang selamat (Survived=1)? gunakan oprasi Numphy

In [7]:
round(df["Survived"].mean() * 100, 2)

np.float64(38.38)

presentase penumpang yang selamat adalah sekitar 38% dari total peenumpang

Q4: Filter penumpang wanita dari kelas 1 (Pclass=1). Berapa jumlahnya dan berapa rata-rata usianya?

In [9]:
wanita_kelas1 = df[(df["Sex"] == "female") & (df["Pclass"] == 1)]

print("Jumlah", len(wanita_kelas1))
print("Rata-rata umur:", wanita_kelas1["Age"].mean())

Jumlah 94
Rata-rata umur: 34.61176470588235


terdapat 94 penumpang wanita pada kelas 1 dengan rata-rata usia sekitar 34.6 tahun

Q5: Gunakan groupby untuk menghitung tingkat keselamatan (mean Survived) per kelas
penumpang. Kelas mana yang paling tinggi?

In [10]:
df.groupby("Pclass")["Survived"].mean()

,Survived
Pclass,
1,0.629630
2,0.472826
3,0.242363


Tingkat keselamatan penumpang berbeda berdasarkan kelas. Penumpang kelas 1 memiliki tingkat keselamatan tertinggi (63%), diikuti kelas 2 (47%), dan kelas 3 (24%). Hal ini menunjukan bahwa kelas sosial mempengaruhi peluang keselamatan.

Q6: Buat analisis tambahan berdasarkan variabel lain yang Anda anggap menarik.
Sertakan interpretasi singkat.

In [11]:
df.groupby("Survived")["Fare"].mean()

,Fare
Survived,
0,22.117887
1,48.395408


Berdasarkan analisis harga tiket, penumpang yang selamat memiliki rata-rata harga tiket sebesar 48.39, lebih tinggi dibandingkan penumpang yang tidak selamat sebesar 22.11. Hal ini menunjukkan bahwa faktor ekonomi berpengaruh terhadap peluang keselamatan penumpang.